mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/options.scala

```diff
@@ -580,25 +573,34 @@ object OptionGroup extends LazyLogging {
       extends OptionGroup
 
   object Checker extends Configurable[Config.Checker, Checker] with LazyLogging {
-    def apply(checker: Config.Checker): Try[Checker] = for {
-      algo <- checker.algo.toTry("checker.algo")
-      discardDisabled <- checker.discardDisabled.toTry("checker.discardDisabled")
-      length <- checker.length.toTry("checker.length")
-      maxError <- checker.maxError.toTry("checker.maxError")
-      noDeadlocks <- checker.noDeadlocks.toTry("checker.noDeadlocks")
-      nworkers <- checker.nworkers.toTry("checker.nworkers")
-      smtEncoding <- checker.smtEncoding.toTry("checker.smtEncoding")
-      tuning <- checker.tuning.toTry("checker.tuning")
-    } yield Checker(
-        algo = algo,
-        discardDisabled = discardDisabled,
-        length = length,
-        maxError = maxError,
-        noDeadlocks = noDeadlocks,
-        nworkers = nworkers,
-        smtEncoding = smtEncoding,
-        tuning = tuning,
-    )
+    def apply(checker: Config.Checker): Try[Checker] = {
+      val configured = for {
+        algo <- checker.algo.toTry("checker.algo")
+        discardDisabled <- checker.discardDisabled.toTry("checker.discardDisabled")
+        length <- checker.length.toTry("checker.length")
+        maxError <- checker.maxError.toTry("checker.maxError")
+        noDeadlocks <- checker.noDeadlocks.toTry("checker.noDeadlocks")
+        nworkers <- checker.nworkers.toTry("checker.nworkers")
+        smtEncoding <- checker.smtEncoding.toTry("checker.smtEncoding")
+        tuning <- checker.tuning.toTry("checker.tuning")
+      } yield Checker(
+          algo = algo,
+          discardDisabled = discardDisabled,
+          length = length,
+          maxError = maxError,
+          noDeadlocks = noDeadlocks,
+          nworkers = nworkers,
+          smtEncoding = smtEncoding,
+          tuning = tuning,
+      )
+      // do some validation
+      configured.foreach(c =>
+        if (checker.view.isEmpty && c.maxError > 1) {
+          throw new PassOptionException(s"Option maxError = ${c.maxError} requires a view. None specified.")
+        })
+      // produce the configured checker
+      configured
+    }
   }
```

Comment:
I think the way the validation logic has been added here is more invasive than necessary and makes it harder to reason about the validation logic. 

I've two main points in mind:

1. Each binding in the `for` comprehension is already a validation step, and since we are here validating the value for `maxError`, our validation can be focused on that one binding.
2. The `apply` method is already managing errors via the `Try`, so throwing an exception makes the error handling more complex. Instead, we can use the try to put the throwable in a `Failure`.

Those lead me to suggest this alternative:

```suggestion
    def apply(checker: Config.Checker): Try[Checker] = for {
      algo <- checker.algo.toTry("checker.algo")
      discardDisabled <- checker.discardDisabled.toTry("checker.discardDisabled")
      length <- checker.length.toTry("checker.length")
      noDeadlocks <- checker.noDeadlocks.toTry("checker.noDeadlocks")
      nworkers <- checker.nworkers.toTry("checker.nworkers")
      smtEncoding <- checker.smtEncoding.toTry("checker.smtEncoding")
      tuning <- checker.tuning.toTry("checker.tuning")
      maxError <- checker.maxError.toTry("checker.maxError").flatMap { n =>
        if (n > 1 && checker.view.isEmpty) {
          val msg = s"Option maxError = ${n} requires a view. None specified."
          Failure(new PassOptionException(msg))
        } else {
          Success(n)
        }
      }
    } yield Checker(
        algo = algo,
        discardDisabled = discardDisabled,
        length = length,
        maxError = maxError,
        noDeadlocks = noDeadlocks,
        nworkers = nworkers,
        smtEncoding = smtEncoding,
        tuning = tuning,
    )
```

I think this is fairly readable, but it could perhaps be improved further with a helper function.

Reply:
True. I had no idea how to do that properly. @shonfeder could you maybe take over this PR. I feel like I am losing the context way too fast :)

Reply:
Yep, sure! Happy to make the changes here.